### ISSUE TITLE
Incorrectly removes generics making the code uncompileable

### ISSUE THREAD

**Issue by dankinsoid** [nicklockwood/SwiftFormat#1392]
I have a generic struct and I need methods with condition  `where Value == Optional<T>`
```swift
@dynamicMemberLookup
public struct Ref<Value> {
}

public extension Ref {

    static func weak<Base: AnyObject, T>(
        _ base: Base,
        _ keyPath: ReferenceWritableKeyPath<Base, Value>
    ) -> Ref<Value> where T? == Value {
    }
}
```
It works well (I often use this workaround for optionals), but swiftformatter removes `T` generic and brokes my code, I didn't find any rule to turn it off
```swift
    static func weak<Base: AnyObject>(
        _ base: Base,
        _ keyPath: ReferenceWritableKeyPath<Base, Value>
) -> Ref<Value> {
}
```

**Comment by nicklockwood** [comment]
Thanks for reporting this. The rule you need to disable is `opaqueGenericParameters`.

**Comment by nicklockwood** [comment]
For future reference running with `--verbose` or `--lint` mode will tell you which rules were applied (and therefore which you need to disable).

**Comment by nicklockwood** [comment]
@calda would you mind taking a look at this? (No problem if you don't have time)

**Comment by calda** [comment]
Sure, I'll take a look.

**Comment by nicklockwood** [comment]
@dankinsoid fix landed in 0.51.3
